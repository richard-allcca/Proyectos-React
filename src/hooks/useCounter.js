import { useState } from "react";

const useCounter = (initalState = 10) => {
  const [state, setState] = useState(initalState);
  const incrementar = (factor = 1) => {
    setState(state + factor);
  };
  const decrementar = (factor = 1) => {
    setState(state - factor);
  };
  const reset = () => {
    setState(initalState);
  };
  return { state, incrementar, decrementar, reset };
};

export default useCounter;
