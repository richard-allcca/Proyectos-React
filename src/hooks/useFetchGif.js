import { useEffect, useState } from "react";
// import getGif from "../helpers/searchGif";

export const useFetchGif = (url) => {
  const [data, setData] = useState(null);
  const [isPending, setIsPending] = useState(true);
  const [error, setError] = useState(null);

  useEffect(() => {
    const getData = async () => {
      try {
        let res = await fetch(url);

        if (!res.ok) {
          let errorThorw = {
            err: true,
            status: res.status,
            statusText: !res.statusText ? "Ocurrio un error" : res.statusText,
          };
          throw errorThorw;
        }

        let datos = await res.json();
        let { data } = datos; //! agregado solo para usar con gif-app
        setIsPending(false);
        setError({ error: false });
        setData(data);
      } catch (error) {
        setError(error);
        setIsPending(false);
      }
    };
    getData(url);
  }, [url]);

  return { data, isPending, error };
};

//! metodo de fernando
// export const useFetchGif = (category) => {
//   const [state, setState] = useState({
//     data: [],
//     loading: true,
//   });

//   useEffect(() => {
//     getGif(category).then((data) => {
//       setState({
//         data,
//         loading: false,
//       });
//     });
//   }, [category]);

//   return state;
// };
