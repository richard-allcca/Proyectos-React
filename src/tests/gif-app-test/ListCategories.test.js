import { shallow } from "enzyme";
import React from "react";
import { useFetchGif } from "../../hooks/useFetchGif";
import ListCategories from "../../proyectos/gif-app/gif-components/ListCategories";
jest.mock("../../hooks/useFetchGif");

describe("Pruebas en el <ListCategories/>", () => {
  const category = "one punch";
  let wrapper = shallow(<ListCategories category={category} />);

  beforeEach(() => {
    wrapper = shallow(<ListCategories category={category} />);
  });

  // ============================
  test("Debe mostrar el contenido de <ListCategories>", () => {
    useFetchGif.mockReturnValue({
      data: [],
      isPending: true,
      // error: false,
    });

    expect(wrapper).toMatchSnapshot();
  });

  // ============================
  test("Debe mostrar la data de useFetchGif", () => {
    const returnUseFetchGif = [
      {
        id: "asd",
        url: "http://localhost/loquesea",
        title: "El Contenido",
      },
    ];

    useFetchGif.mockReturnValue({
      data: returnUseFetchGif,
      isPending: false,
      // error: false,
    });
  });
});

// Notas:
// jest.mock(''): simula la llamada a ese archivo y controlar la info que este responde
// el 2Â° test no es posible porque envio como parametro la 'url' y el envia un 'string'
