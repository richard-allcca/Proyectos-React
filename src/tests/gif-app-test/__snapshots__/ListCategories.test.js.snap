// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <ListCategories/> Debe mostrar el contenido de <ListCategories> 1`] = `""`;
